File: frontend/src/api/models/EnterpriseOwnerUserCreateRequestDTO.ts
import type { Account } from './Account.ts'
import type { EnterpriseUnit } from './EnterpriseUnit.ts'

export const enterpriseOwnerUserCreateRequestDTODocumentTypeEnum = {
  CPF: 'CPF',
  CNPJ: 'CNPJ',
} as const

export type EnterpriseOwnerUserCreateRequestDTODocumentTypeEnum =
  (typeof enterpriseOwnerUserCreateRequestDTODocumentTypeEnum)[keyof typeof enterpriseOwnerUserCreateRequestDTODocumentTypeEnum]

export type EnterpriseOwnerUserCreateRequestDTO = {
  /**
   * @type string | undefined
   */
  name?: string
  /**
   * @type string | undefined
   */
  rawDocumentNumber?: string
  /**
   * @type string | undefined
   */
  documentType?: EnterpriseOwnerUserCreateRequestDTODocumentTypeEnum
  /**
   * @type object | undefined
   */
  account?: Account
  /**
   * @type array | undefined
   */
  units?: EnterpriseUnit[]
}

File: backend/src/main/java/br/com/visitsafe/dto/user/EnterpriseOwnerUserCreateRequestDTO.java
package br.com.visitsafe.dto.user;

import br.com.visitsafe.model.Account;
import br.com.visitsafe.model.enums.DocumentTypeEnum;
import br.com.visitsafe.model.unit.EnterpriseUnit;
import lombok.Data;

import java.util.List;

@Data
public class EnterpriseOwnerUserCreateRequestDTO {
    private String name;
    private String rawDocumentNumber;
    private DocumentTypeEnum documentType;
    private Account account;
    private List<EnterpriseUnit> units;
}
